test/fixtures/dummy_distro_plugin/bin/list-all

```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+versions_list=(acme-1.0.0 acme-1.1.0 distro-2.0.0)
```

Comment:
We should also test versions which have multiple prefix parts, EG: `zulu-jre-musl-8.42.0.23`

More examples from `asdf python`

```
pypy3.9-7.3.8
pypy3.9-7.3.9-src
pypy3.9-7.3.9
pyston-2.2
pyston-2.3
pyston-2.3.1
pyston-2.3.2
pyston-2.3.3
pyston-2.3.4
stackless-dev
stackless-2.7-dev
```

Reply:
```
pypy3.9-7.3.8
pypy3.9-7.3.9-src
```

IMO these are two different variants, though most querying added to commands are prefix-only queries, so I guess these should be resolved as a single variant.

Reply:
Good point, current implemenation does not yet handle suffixes. But i agree that this example should be treatet as a single variant. I would consider suffixes after the sematic version not as a variant but rather a special version like beta, GA, src or similar.